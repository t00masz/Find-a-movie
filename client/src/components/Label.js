import React from 'react';

const Label = (props) => (
  <div className="option">
    <label name={props.name} 
      className="option__label">
        {props.value}
    </label>
  </div>
)

export default Label;