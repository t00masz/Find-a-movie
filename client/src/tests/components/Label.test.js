import React from 'react';
import { shallow } from 'enzyme'
import Label from '../../components/Label';

test('should render label', () => {
    const wrapper = shallow(<Label 
        name="errorLabel" 
        value="Testing label" />);
    expect(wrapper).toMatchSnapshot();
});
