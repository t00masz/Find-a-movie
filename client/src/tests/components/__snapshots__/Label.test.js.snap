// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    name="errorLabel"
    value="Testing label"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="option__label"
        name="errorLabel"
>
        Testing label
</label>,
      "className": "option",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Testing label",
        "className": "option__label",
        "name": "errorLabel",
      },
      "ref": null,
      "rendered": "Testing label",
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="option__label"
          name="errorLabel"
>
          Testing label
</label>,
        "className": "option",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Testing label",
          "className": "option__label",
          "name": "errorLabel",
        },
        "ref": null,
        "rendered": "Testing label",
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
